tox.ini

```diff
@@ -1,13 +1,18 @@
 [tox]
-envlist={py27,py34,py35,py36}-django{20,30}
+envlist={py38,py39,py310,py311}-{django22,django32,django42}
 
 [testenv]
+allowlist_externals=pytest
 basepython=
-    py35: python3.5
-    py36: python3.6
-    py37: python3.7
     py38: python3.8
+    py39: python3.9
+    py310: python3.10
+    py311: python3.11
 deps=
-    django20: django==2.0
-    django30: django==3.0
-commands=python setup.py test
+    django22: django==2.2.28
+    django32: django==3.2.22
+    django42: django==4.2.6
+extras = test
+commands =
+  pytest
+  black --check .
```

Comment:
Should this be `pre-commit run -a` instead of just Black?

Reply:
That will also run `gitchangelog`. From what I can gather of what that package does, that would not be desirable in a CI environment as it'll change the state of the repo. Proceed anyway?

(on the other hand, that did just find an unused import in `conftest.py`...)

Reply:
You're right, the correct command is `pre-commit run -a --hook-stage=manual`, which skips gitchangelog.

Reply:
That should be fixed in 3c8d7997. And it found that my imports were in the wrong order there, which is nice!